### ISSUE TITLE
method error in QCBM.ipynb 

### ISSUE THREAD

**Issue by LionSR** [QuantumBFS/Yao.jl#125]
My Julia version is Julia0.6.4. I tried to run the QCBM.ipynb example directly. And the following codes invokes an method error.

```julia
optim = Adam(lr=0.1)
his = train!(circuit, pg, optim, niter=50, learning_rate=0.1)
plot(1:50, his, xlabel="iteration", ylabel="loss")
```
The error message is 

> MethodError: Cannot `convert` an object of type IterTools.Chain{NTuple{186,Float64}} to an object of type Int64
This may have arisen from a call to the constructor Int64(...),
since type constructors fall back to convert methods.
Stacktrace:
 [1] update!(::IterTools.Chain{NTuple{186,Float64}}, ::Array{Float64,1}, ::Adam) at ./In[28]:16
 [2] #train!#16(::Float64, ::Int64, ::Function, ::Yao.Blocks.ChainBlock{6,Complex{Float64}}, ::Array{Float64,1}, ::Adam) at ./In[29]:15
 [3] (::#kw##train!)(::Array{Any,1}, ::#train!, ::Yao.Blocks.ChainBlock{6,Complex{Float64}}, ::Array{Float64,1}, ::Adam) at ./<missing>:0

It would be great help if this is fixed. Great works!

**Comment by Roger-luo** [comment]
probably because I forgot to update the example with new versions of Yao. I'll fix this.

**Comment by Roger-luo** [comment]
I just tag a deprecation fix for Julia 1.0 (v0.6 support is drop) as `v0.2.0`, try it with

```julia
pkg> add Yao#v0.2.0
```

or wait for https://github.com/JuliaLang/METADATA.jl/pull/16898 and simply update your old package

```julia
pkg> up Yao
```

Or just use the master branch (which may not be stable)

```
pkg> add Yao#master
```
